src/server/protocol_client.cc

```diff
@@ -228,6 +228,10 @@ error_code ProtocolClient::ResolveMasterDns() {
   return error_code{};
 }
 
+void ProtocolClient::InitEndpoint() {
+  server_context_.endpoint = {ip::make_address(server_context_.host), server_context_.port};
```

Comment:
It is confusing that sometimes server_context_.host is the ip and sometimes its the host name.
I see there is constructor for ProtocolClient with server_context that can you can skip the Dns resolution so I believe you dont need this changes in ProtocolClient api